Jenkinsfile

```diff
@@ -70,103 +57,190 @@ pipeline {
+          not { expression { isPr } }
+          not { expression { return params.RELEASE_BUILD && params.RELEASE_DRY_RUN } }
+        }
       }
-      steps {
-        script {
-          lastStage = env.STAGE_NAME
-          // Here's where we're getting the secondary artifact (named
-          // via env.STORE_PG_ARTIFACT) out for dockerizing, per the
-          // comment above.
-          if (params.RELEASE_BUILD) {
-            env.REGISTRY_PUSH = (params.RELEASE_DRY_RUN) ? 'none' : 'all'
-            env.SECONDARY_DOCKER_TAG = dockerize_secondary.docker_build_specify_tag_and_push(params.RELEASE_NAME, sbtbuild.getDockerPath(project_wd_secondary), env.STORE_PG_ARTIFACT, env.REGISTRY_PUSH)
-          } else {
-            env.REGISTRY_PUSH = 'internal'
-            env.SECONDARY_DOCKER_TAG = dockerize_secondary.docker_build('STAGING', env.GIT_COMMIT, sbtbuild.getDockerPath(project_wd_secondary), env.STORE_PG_ARTIFACT, env.REGISTRY_PUSH)
+      parallel {
+        stage('Publish - SoQL Server') {
+          steps {
+            script {
+              lastStage = env.STAGE_NAME
+              if (params.RELEASE_BUILD) {
+                env.BUILD_ID = dockerize_server.publish(sourceTag: env.DOCKER_TAG)
+              } else {
+                env.BUILD_ID = dockerize_server.publish(
+                  sourceTag: env.DOCKER_TAG,
+                  environments: ['internal']
+                )
+              }
+            }
+          }
+          post {
+            success {
+              script {
+                if (params.RELEASE_BUILD) {
+                  Map buildInfo = [
+                    "project_id": service_server,
+                    "build_id": env.BUILD_ID,
+                    "release_id": params.RELEASE_NAME,
+                    "git_tag": env.GIT_TAG
+                  ]
+                  createBuild(
+                    buildInfo,
+                    rmsSupportedEnvironment.production
+                  )
+                }
+              }
+            }
+          }
+        }
+        stage('Publish - Secondary') {
+          steps {
+            script {
+              lastStage = env.STAGE_NAME
+              if (params.RELEASE_BUILD) {
+                env.SECONDARY_BUILD_ID = dockerize_secondary.publish(sourceTag: env.SECONDARY_DOCKER_TAG)
+              } else {
+                env.SECONDARY_BUILD_ID = dockerize_secondary.publish(
+                  sourceTag: env.SECONDARY_DOCKER_TAG,
+                  environments: ['internal']
+                )
+              }
+            }
+          }
+          post {
+            success {
+              script {
+                if (params.RELEASE_BUILD) {
+                  Map buildInfo = [
+                    "project_id": service_secondary,
+                    "build_id": env.SECONDARY_BUILD_ID,
+                    "release_id": params.RELEASE_NAME,
+                    "git_tag": env.GIT_TAG
+                  ]
+                  createBuild(
```

Comment:
With respect to notifying RMS, what happens if one of the artifacts (soql-server, secondary) fails and the other succeeds?

Reply:
The stages are serial, so if there is a failure in the docker build stage, then it won't notify RMS.  
If publishing soql-server succeeds, but not secondary, then we have code problems in the pipelines repo.  Tactically, we can rebuild with `replay` and select which stages to run.